accord-core/src/test/java/accord/utils/Gen.java

```diff
@@ -69,6 +89,16 @@ default Gen<A> filter(Predicate<A> fn)
         };
     }
 
+    default Supplier<A> asSupplier(RandomSource rs)
+    {
+        return () -> next(rs);
+    }
+
+    default Stream<A> asStream(RandomSource rs)
+    {
+        return Stream.generate(() -> next(rs));
+    }
+
     interface IntGen extends Gen<Integer>
```

Comment:
Why does this extend `Gen<Integer>`? This leads to much uglier mappings, and confusion about whether there is any boxing. If this utility is to be more broadly used, it would be nice to simplify the reading and writing of test cases with it. If we want to box or unbox, a `boxed()` method is preferable.

Reply:
> Why does this extend Gen<Integer>?

Its so you can still use it in all contexts, by having disjoint interfaces that have the same shape, it gets harder to compose together.

>  If this utility is to be more broadly used, it would be nice to simplify the reading and writing of test cases with it

Can you expand on this?  Several tests use this and the `LongGen` version.

> If we want to box or unbox, a boxed() method is preferable.

I am +0 to `.boxed()` vs extends; they solve the same problem.  I can test this out but think I would need to add more methods to this type and gets easier for these primitive types to miss methods added to the base.